HacktoberfestProject.Web/Services/TableService.cs

```diff
@@ -49,11 +49,46 @@ public async Task<ServiceResponse<User>> GetUserAsync(string username)
 
             if (entities.Any())
             {
-                var user = new User(username);
-                foreach (var entity in entities.OrderBy(e => e.RowKey))
+                var user = new User(username, new List<Repository>());
+                foreach (var entity in entities)
                 {
+                    bool add = false;
                     var info = entity.RowKey.Split(':', StringSplitOptions.RemoveEmptyEntries);
-                    // TODO: implement splitting the entities into a class-structure for the frontend model
+                    if (info.Length != 3)
+                    {
+                        // TODO: define how to proceed with this.. this is corrupted data!
+                        continue;
+                    }
+                    var owner = info[0];
+                    var repoName = info[1];
+                    var prId = int.Parse(info[2]);
+
+                    Repository repo;
+                    var temp = user.RepositoryPrAddedTo
+                                   .FirstOrDefault(repo => repo.Owner.Equals(owner, StringComparison.OrdinalIgnoreCase) &&
+                                                           repo.Name.Equals(repoName, StringComparison.OrdinalIgnoreCase));
+                    if (temp != null)
+                    {
+                        repo = temp;
+                    }
+                    else 
+                    {
+                        repo = new Repository(owner, repoName, entity.Url, new List<PullRequest>());
+                        add = true;
+                    }
+                    var prs = entities.Where(e => e.RowKey.StartsWith($"{owner}:{repoName}", StringComparison.OrdinalIgnoreCase));
+                    foreach (var pr in prs)
+                    {
+                        var id = int.Parse(pr.RowKey.Substring(pr.RowKey.LastIndexOf(':') + 1));
+                        if (!repo.Prs.Any(pr => pr.PrId == id))

```

Comment:
There is a lot of code happening in this method. I would like it decomposed into smaller units to keep it all more solid.